src/main/java/com/google/devtools/build/lib/skyframe/LocalDiffAwareness.java

```diff
@@ -105,7 +107,10 @@ public DiffAwareness maybeCreate(Root pathEntry) {
         return new MacOSXFsEventsDiffAwareness(resolvedPathEntryFragment.toString());
       }
 
-      return new WatchServiceDiffAwareness(resolvedPathEntryFragment.toString());
+      return new WatchServiceDiffAwareness(
+        resolvedPathEntryFragment.toString(),
+        ignorePaths.stream().map(p -> p.toString()).collect(toImmutableSet())
```

Comment:
Why do we convert `ImmutableSet<Path>` to `ImmutableSet<String>` here, and then for each of the application of `ignorePaths` in `WatchServiceDiffAwareness`, we convert it to `ImmutableSet<Path>` again? I think we should just store `ignorePaths` as `ImmutableSet<Path>`.

Reply:
This seems reasonable. I think there will still be one layer of conversion from `com.google.devtools.build.lib.vfs.Path` to `java.nio.file.Path` because `WatchServiceDiffAwareness` and related files don't use `com.google.devtools.build.lib.vfs` at all and it feels like a violation of the layering to have them use it.